src/message/MessageContainer.js

```diff
@@ -87,21 +88,38 @@ class MessageContainer extends PureComponent {
     this.showActionSheet({ options, cancelButtonIndex: options.length - 1, callback });
   };
 
+  handleOutboxMessageLongPress = () => {
+    const { actions, message } = this.props;
+    const callback = buttonIndex => {
+      switch (buttonIndex) {
+        case 0: {
+          actions.deleteOutboxMessage(message.timestamp);
+          break;
+        }
+        default:
+          break;
+      }
+    };
+    const options = ['Delete message', 'Cancel'];
+    this.showActionSheet({ options, cancelButtonIndex: options.length - 1, callback });
+  };
+
   render() {
-    const { message, auth, actions, twentyFourHourTime, isBrief } = this.props;
+    const { message, auth, actions, twentyFourHourTime, isBrief, isOutbox } = this.props;
     const MessageComponent = isBrief ? MessageBrief : MessageFull;
 
     return (
       <MessageComponent
         message={message}
         twentyFourHourTime={twentyFourHourTime}
         ownEmail={auth.email}
-        onLongPress={this.handleLongPress}
-        starred={this.isStarred(message)}
+        onLongPress={isOutbox ? this.handleOutboxMessageLongPress : this.handleLongPress}
+        starred={isOutbox ? undefined : this.isStarred(message)}
         realm={auth.realm}
         auth={auth}
         actions={actions}
         handleLinkPress={this.handleLinkPress}
+        isOutbox={isOutbox}
       />
     );
   }
```

Comment:
You need to think more in the direction of 'separation of concerns'.
Adding more and more logic to a single file/component usually breaks it.
MessageContainer ideally should have no idea about the outbox concept.
In fact, outbox should be as similar to non-outbox as possible, so very little 'ifs' are needed.